eng/common/TestResources/New-TestResources.ps1

```diff
@@ -200,35 +200,62 @@ try {
-                'a18897a6-7e44-457d-9260-f2854c0aca42' = 'Azure SDK Engineering System'
-                '2cd617ea-1866-46b1-90e3-fffb087ebf9b' = 'Azure SDK Test Resources'
+            # Use the given subscription ID if provided.
+            $subscriptionArgs = if ($SubscriptionId) {
+                @{Subscription = $SubscriptionId}
+            } else {
+                @{}
             }
 
-            if ($wellKnownSubscriptions.ContainsKey($SubscriptionId)) {
-                $subscriptionName = '{0} ({1})' -f $wellKnownSubscriptions[$SubscriptionId], $SubscriptionId
+            # TODO: Should we default DeleteAfterHours for the Playground sub to 48hrs or even a week? Could really help with costs.
+            # Or should we prompt - default and ask to opt out of DeleteAfterHours?
+
+            Log 'Logging you in...'
+            $context = (Connect-AzAccount @subscriptionArgs).Context
+
+            # Warn that a "random" subscription was selected.
+            if (!$SubscriptionId) {
+                $subscriptionName = $context.Subscription.Id
+
+                # Use cache of well-known team subs without having to be authenticated.
+                $wellKnownSubscriptions = @{
+                    'faa080af-c1d8-40ad-9cce-e1a450ca5b57' = 'Azure SDK Developer Playground'
+                    'a18897a6-7e44-457d-9260-f2854c0aca42' = 'Azure SDK Engineering System'
+                    '2cd617ea-1866-46b1-90e3-fffb087ebf9b' = 'Azure SDK Test Resources'
+                }
+
+                if ($wellKnownSubscriptions.ContainsKey($subscriptionName)) {
+                    $subscriptionName = '{0} ({1})' -f $wellKnownSubscriptions[$subscriptionName], $subscriptionName
+                }
+
+                Write-Warning "No subscription was specified, so $subscriptionName was automatically selected."
             }
+        }
+
+        # Make sure subscription and tenant IDs are set.
+        if (!$SubscriptionId) {
+            $SubscriptionId = $context.Subscription.Id
+            $PSBoundParameters['SubscriptionId'] = $SubscriptionId
+        }
 
-            Log "You are not logged in; connecting to $subscriptionName"
-            $context = (Connect-AzAccount -Subscription $SubscriptionId).Context
+        if (!$TenantId) {
+            $TenantId = $context.Subscription.TenantId
         }
 
         # If no test application ID is specified during an interactive session, create a new service principal.
         if (!$TestApplicationId) {
 
             # Cache the created service principal in this session for frequent reuse.
-            $servicePrincipal = if ($AzureTestPrincipal -and (Get-AzADServicePrincipal -ApplicationId $AzureTestPrincipal.ApplicationId)) {
+            $servicePrincipal = if ($AzureTestPrincipal -and (Get-AzADServicePrincipal -ApplicationId $AzureTestPrincipal.ApplicationId) -and $AzureTestSubscription -eq $SubscriptionId) {
```

Comment:
Doesn't AzureTestPrincipal have a subscription id property?

Reply:
Surprisingly, no. I was planning on using that originally.
```
PS 90  00:00:00.009  C:\src\azure-sdk-for-net 
  test-resources  $AzureTestPrincipal | gm

   TypeName: Microsoft.Azure.Commands.Resources.Models.Authorization.PSADServicePrincipalWrapper

Name                  MemberType Definition
----                  ---------- ----------
Equals                Method     bool Equals(System.Object obj)
GetHashCode           Method     int GetHashCode()
GetType               Method     type GetType()
ToString              Method     string ToString()
ApplicationId         Property   guid ApplicationId {get;set;}
DisplayName           Property   string DisplayName {get;set;}
Id                    Property   string Id {get;set;}
ObjectType            Property   string ObjectType {get;}
Secret                Property   securestring Secret {get;set;}
ServicePrincipalNames Property   string[] ServicePrincipalNames {get;set;}
Type                  Property   string Type {get;set;}

PS 91  00:00:00.020  C:\src\azure-sdk-for-net 
  test-resources  $AzureTestPrincipal | select *

Secret                : System.Security.SecureString
ServicePrincipalNames : {c8c0de1d-ccd8-4f84-ba8c-9db4138833c4, http://azure-powershell-03-30-2021-01-19-25}
ApplicationId         : c8c0de1d-ccd8-4f84-ba8c-9db4138833c4
ObjectType            : ServicePrincipal
DisplayName           : azure-powershell-03-30-2021-01-19-25
Id                    : d3c406a8-21c6-4eaa-bbfa-c5070e7cb3a7
Type                  :
```